File: 02x10/c-019/main.rb
require 'date'

def numberOfSundays
    sundays = 0
    for year in 1901..2000 do
        for month in 1..12 do
            theDay = Date.new(year, month, 1)
            if theDay.wday == 0 then
                sundays += 1
            end
        end
    end
    sundays
end

puts numberOfSundays

File: 02x10/c-019/main.php
<?php

function numberOfSundays() {
    $sundays = 0;
    for ($year = 1901; $year <= 2000; $year++) {
        for ($month = 1; $month <= 12; $month++) {
            $day = mktime(0, 0, 0, $month, 1, $year);
            if (strcmp(date("l", $day), "Sunday") == 0) {
                $sundays++;
            }
        }
    }
    return $sundays;
}

print numberOfSundays();
